crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -703,6 +706,50 @@ impl From<SyncTimelineEventDeserializationHelperV0> for SyncTimelineEvent {
     }
 }
 
+/// Represent a failed to send error of an event sent via the send_queue.
+/// These errors are not automatically retrievable, but yet some manual action
+/// can be taken before retry sending.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Enum))]
+pub enum QueueWedgeError {
+    /// This error occurs when there are some insecure devices in the room, and
+    /// the current encryption setting prohibit sharing with them.
+    InsecureDevices {
+        /// The insecure devices as a Map of userID to deviceID.
+        user_device_map: HashMap<String, Vec<String>>,
+    },
+    /// This error occurs when a previously verified user is not anymore, and
+    /// the current encryption setting prohibit sharing when it happens.
+    IdentityViolations {
+        /// The users that are expected to be verified but are not.
+        users: Vec<String>,
+    },
+    /// It is required to set up cross-signing and properly verify the current
+    /// session before sending.
+    CrossVerificationRequired,
+    /// Other errors.
+    GenericApiError { msg: String },
```

Comment:
Could it be a `Box<dyn Error>` or something similar? It's a bit weird to store the stringified error, since API users can't match on it, unless they precisely know what string to expect.

Reply:
see here https://github.com/matrix-org/matrix-rust-sdk/pull/4137#issuecomment-2425914825